python/cudf/cudf/io/parquet.py

```diff
@@ -609,11 +609,12 @@ def _parquet_to_frame(
         )
         # Add partition columns to the last DataFrame
         for (name, value) in part_key:
+            _len = len(dfs[-1])
             if partition_categories and name in partition_categories:
                 # Build the categorical column from `codes`
-                codes = as_column(
-                    partition_categories[name].index(value),
-                    length=len(dfs[-1]),
+                codes = full(
+                    size=_len,
+                    fill_value=partition_categories[name].index(value),
```

Comment:
Yes, exactly. We are reading a file from a hive-partitioned dataset. This means we are filling every row in the hive-partitioned column with the same value (a value that was encoded in the name of the directory where the file was stored). For example, if we were reading from the following path `"root_dir/month=12/part.2.parquet"`, we would need to assign `12` (or the categorical equivalent) to every row of the `"month"` column, since the data would  have been left out of the actual `"part.2.parquet"` file (usually).